<?xml version="1.0" encoding="UTF-8"?>

<launch>
    <!-- Camera -->
    <!-- <node name="usb_cam" pkg="usb_cam" type="usb_cam_node">
        <param name="video_device" value="/dev/video0" />
        <param name="autoexposure" value="false" />
        <param name="pixel_format" value="yuyv" />
        <param name="camera_frame_id" value="usb_cam" />
        <param name="io_method" value="mmap" />
        
        <param name="image_width" value="640" />
        <param name="image_height" value="480" />
        <param name="exposure" value="10" />
        <param name="contrast" value="0" />
        <param name="saturation" value="100" />
    </node> -->

    <!-- Recording -->
    <node pkg="rosbag" type="play" name="rosbag_play" required="true" args="$(find tricat221)/data/rosbag/docking.bag -l --topics /imu/mag /ublox_gps/fix /usb_cam/image_raw "/>
    
    <!-- Nodes -->
    <node pkg="tricat221"  type="heading_calculator.py" name="headingCalculator"/>
	<node pkg="tricat221"  type="gnss_converter.py" name="gnssConverter"/>
	<node pkg="tricat221" type="lidar_converter.py" name="lidarConverter" output="screen"/>
	<node pkg="tricat221" type="docking.py" name="Docking" output="screen"  respawn="true"/>

    <!-- Visualization -->
	<!-- <node name="rqt" pkg="rqt_graph" type="rqt_graph"/>	 -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find tricat221)/rviz/rviz_conf_docking.rviz" />

    <!-- Parameters -->
	<rosparam command="load" file="$(find tricat221)/params/lidar_params.yaml"/>
	<rosparam command="load" file="$(find tricat221)/params/docking_params.yaml"/>
	<rosparam command="load" file="$(find tricat221)/params/coordinates.yaml"/>

</launch>